### ISSUE TITLE
CMake find_package warnings

### ISSUE THREAD

**Issue by martukas** [ess-dmsc/h5cpp#220]
I'm getting this in project that use h5cpp. Since it says `This warning is for project developers`, I guess that means us? ;)
```
CMake Warning (dev) at /home/bdv/.conan/data/h5cpp/0.0.4-dm1/ess-dmsc/stable/package/485c38981996db847b6ae4e6a5365f0faa405115/lib/cmake/h5cpp-0.0.4/h5cpp-config.cmake:30 (find_package):
  Ignoring EXACT since no version is requested.
Call Stack (most recent call first):
  prototype2/gdgem/CMakeLists.txt:5 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.
```

**Comment by eugenwintersberger** [comment]
Not really an issue. I guess this comes when you use EXACT in the find_package and provide no version number. Is there one?

**Comment by martukas** [comment]
No, it' just a plain `find_package(h5cpp)` statement. Maybe there is a default setting somewhere that the package comes with.

**Comment by martukas** [comment]
I have added Afonso to the discussion. Maybe he has some insight on this.

**Comment by amues** [comment]
Line 30 in _h5cpp-config.cmake_ is causing this:
```
find_package(HDF5  EXACT REQUIRED COMPONENTS C)
```
This file is generated by h5cpp and installed to _<prefix>/lib/cmake/_

**Comment by eugenwintersberger** [comment]
Ah. Thank you guys. I ssee the point. This requires a fix. Basically we should pin the HDF5 version here in order to ensure binary compatibility.

**Comment by eugenwintersberger** [comment]
This should be already fixed in the current master. I guess I simply forgot to close the tickt.

**Comment by martukas** [comment]
This is still not fixed as of v.0.0.5.
The situation described in comment by @amues still holds true. I guess, for some reason `@HDF5_VERSION@` is rendered as empty. Is this something that only happens on some systems? Can we test for this in CI? Is this something related to how the conan package is generated?

**Comment by martukas** [comment]
I just checked, and this variable is definitely not rendered in my local build on Ubuntu17. So this is definitely an `h5cpp` issue and not a conan recipe one.

**Comment by eugenwintersberger** [comment]
I do not see this issue on Debian 9 and 8. Hm. That's a bit strange.

**Comment by martukas** [comment]
this should have been fixed by #293
